libraries/Microsoft.Bot.Connector/Authentication/MicrosoftAppCredentials.cs

```diff
@@ -31,24 +32,20 @@ public class MicrosoftAppCredentials : ServiceClientCredentials
         /// The key for Microsoft app Password.
         /// </summary>
         public const string MicrosoftAppPasswordKey = "MicrosoftAppPassword";
+        
+        private static HttpClient _httpClient = new HttpClient();
 
-        /// <summary>
-        /// The token refresh code uses this client. Ideally, this would be passed in or set via a DI system to 
-        /// allow developer control over behavior / headers / timeouts and such. Unfortunately this is buried
-        /// pretty deep, the static solution used here is much cleaner. If this becomes an issue we could
-        /// consider circling back and exposing developer control over this HttpClient. 
-        /// </summary>
-        private static readonly HttpClient _httpClient = new HttpClient();
-
-        private static object _trustedHostNamesSync = new object();
-        private static readonly IDictionary<string, DateTime> _trustedHostNames = new Dictionary<string, DateTime>()
-        {
-            { "state.botframework.com", DateTime.MaxValue },
-            { "api.botframework.com", DateTime.MaxValue}
-        };
+        protected static ConcurrentDictionary<string, DateTime> TrustedHostNames = new ConcurrentDictionary<string, DateTime>(
+                                                                                        new Dictionary<string, DateTime>() {
+                                                                                            { "state.botframework.com", DateTime.MaxValue },
```

Comment:
Are we still supporting state.botframework.com? Would now be the time to remove that?